clang/include/clang/AST/PreorderAST.h

```diff
@@ -25,24 +25,27 @@ namespace clang {
   using Result = Lexicographic::Result;
 
   // Each binary operator of an expression results in a new node of the
-  // PreorderAST. Each node contains 3 fields:
+  // PreorderAST. Each node contains the following fields:
+
   // Opc: The opcode of the operator.
   // Vars: A list of variables in the sub expression.
   // Const: Constants of the sub expression are folded.
+  // HasConst: Indicates whether there is a constant in the node. It is used to
+  // differentiate between the absence of a constant and a constant value of 0.
+  // Parent: A link to the parent node of the current node.
+  // Children: The preorder AST is an n-ary tree. Children is a list of all the
+  // child nodes of the current node.
 
   struct Node {
     BinaryOperator::Opcode Opc;
     std::vector<const VarDecl *> Vars;
     llvm::APSInt Const;
-    // HasConst indicates whether there is a constant in the node. This is used
-    // to differentiate between an absence of a constant and a constant of value
-    // 0.
     bool HasConst;
-    Node *Parent, *Left, *Right;
+    Node *Parent;
+    llvm::SetVector<Node *> Children;
```

Comment:
How about using a SmallVector instead?   Why use a SetVector?

Reply:
I used a SetVector to reduce make it easier to remove an element from the list. But SetVector has a larger space complexity than a vector. I will change this to a SmallVector.